### PR TITLE
Grid Printer exercise complete

### PR DESCRIPTION
**Pull request by joejohnsto**
see students/joejohnsto/lesson02

lesson02 grid printer complete
also first commit with lesson01 (break_me and pushups1)
and with lesson02 pushups2

### CODE REVIEW

**Comment by joejohnsto** [comment]
submittal for lesson02 grid printer exercise

**Comment by joejohnsto** [comment]
second commit add the FizzBuzz exercise to the pull request

**Comment by joejohnsto** [comment]
Commit with the series exercise complete

**Comment by joejohnsto** [comment]
lesson03 exercises submitted. mailroom.py not complete

File: students/joejohnsto/lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+# Code below calls the functions in order so this file will run like a script
+fruits = list1()
+list2(fruits)
+list3(fruits)
+list4(fruits)
+
+
+if __name__ == "__main__":
```
Comment (majstorpoki) [inline students/joejohnsto/lesson03/list_lab.py:96]: Why?

File: students/joejohnsto/lesson03/mailroom.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+def thank_you():
+    response = 'list'
+    while response == 'list':
+        response = input('What is the full name of the donor to whom you would like to\
+                     send a thank you?\nType "list" for a list of current donors.\n')
+        if response == 'list':
+            for x, y in donor_db:
+                print(x)
```
Comment (majstorpoki) [inline students/joejohnsto/lesson03/mailroom.py:32]: This not completed. Please, read the specs.

Comment (majstorpoki) [reply students/joejohnsto/lesson03/mailroom.py:32]: Also, your functions to add a donor with an amount and to print the list of donors with total amounts, number of donations and the average amount is not completed. If you wish, you can resubmit your work for more points.

Comment (majstorpoki) [reply students/joejohnsto/lesson03/mailroom.py:32]: At this point, I can give you only 5/15 but as I said, you can resubmit the completed assignment (this is not completed, yet)

File: students/joejohnsto/lesson03/slice_lab.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+def reverse(seq):
+    """Return the given sequence in reverse"""
+    return seq[::-1]
+
+def thirds(seq):
+    """Return the given sequence in the order: last third, first third, middle third"""
+    first = seq[:int(len(seq)/3)]
+    last = seq[-int(len(seq)/3):]
+    mid = seq[int(len(seq)/3):-int(len(seq)/3)]
```
Comment (majstorpoki) [inline students/joejohnsto/lesson03/slice_lab.py:34]: Good

File: students/joejohnsto/lesson03/strformat_lab.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+s = f'{t[3]:02} {t[4]} {t[2]} {t[0]:02} {t[1]}'
+
+
+# Task Five
+
+l = ['oranges', 1.3, 'lemons', 1.1]
+s = f'The weight of an {l[0][:-1]} is {l[1]} and the weight of a {l[2][:-1]} is {l[3]}'
+s2 = f'The weight of an {l[0][:-1].upper()} is {l[1]*1.2} and the weight of a {l[2][:-1].upper()} is {l[3]*1.2}'
+
+
```
Comment (majstorpoki) [inline students/joejohnsto/lesson03/strformat_lab.py:44]: Where did you actually PRINT for tasks 1-5?

Comment (majstorpoki) [reply students/joejohnsto/lesson03/strformat_lab.py:44]: Because of that, I deduced you 1 point, but you have 1 extra...

**Review by majstorpoki — APPROVED** [review]

File: students/joejohnsto/lesson03/strformat_lab.py (hunk 4 above)
Comment (joejohnsto) [reply students/joejohnsto/lesson03/strformat_lab.py:44]: I didn't actually print the resultant strings for Task 1-5, but the assignment description does not explicitly state that we needed to print, just that we needed to write code that would build the string correctly.

Comment (majstorpoki) [reply students/joejohnsto/lesson03/strformat_lab.py:44]: I acknowledge this and I am returning back your 1 point. Thanks.

**Comment by majstorpoki** [comment]
Corrected.

Thanks,

David

**Pull request merged**